docker/docker_client.go

```diff
@@ -348,6 +350,7 @@ func SearchRegistry(ctx context.Context, sys *types.SystemContext, registry, ima
 	}
 	client.username = username
 	client.password = password
+	client.registryToken = sys.DockerRegistryToken
```

Comment:
`*types.SystemContext` can always be `nil`, and must be treated the same as `&types.SystemContext{}`.

Reply:
I've fixed it.
https://github.com/containers/image/pull/842/commits/b6597e00d767628a3aed219f53b7daf32d818746

Reply:
This one is still outstanding AFAICS.

Reply:
Oh, I'm sorry. I was confused with https://github.com/containers/image/commit/b6597e00d767628a3aed219f53b7daf32d818746. I've fixed it now. https://github.com/containers/image/pull/842/commits/4e559c3809dacade587698fc446143a0d375b506